src/pg/brc20/brc20-pg-store.ts

```diff
@@ -111,38 +125,38 @@ export class Brc20PgStore {
       FROM events
       INNER JOIN
     `;
+    // todo: use event history
   }
 
   async getTokenSupply(args: { ticker: string }): Promise<DbBrc20Supply | undefined> {
     return await this.parent.sqlTransaction(async sql => {
       const deploy = await this.getDeploy(args);
-      if (!deploy) {
-        return;
-      }
-      const minted = await sql<{ total: string }[]>`
-        SELECT SUM(avail_balance + trans_balance) AS total
-        FROM brc20_balances
+      if (!deploy) return;
+
+      const supplyPromise = sql<{ max: string }[]>`
+        SELECT max FROM brc20_deploys WHERE id = ${deploy.id}
+      `;
+      const mintedPromise = sql<{ minted_supply: string }[]>`
+        SELECT minted_supply
+        FROM brc20_supplies
         WHERE brc20_deploy_id = ${deploy.id}
-        GROUP BY brc20_deploy_id
       `;
-      const holders = await sql<{ count: string }[]>`
-        WITH historical_holders AS (
-          SELECT SUM(avail_balance + trans_balance) AS balance
-          FROM brc20_balances
-          WHERE brc20_deploy_id = ${deploy.id}
-          GROUP BY address
-        )
+      const holdersPromise = sql<{ count: string }[]>`
         SELECT COUNT(*) AS count
-        FROM historical_holders
-        WHERE balance > 0
-      `;
-      const supply = await sql<{ max: string }[]>`
-        SELECT max FROM brc20_deploys WHERE id = ${deploy.id}
+        FROM brc20_balances
+        WHERE brc20_deploy_id = ${deploy.id}
+        GROUP BY address
+        HAVING SUM(avail_balance + trans_balance) > 0
       `;
+      const [supply, holders, minted] = await Promise.all([
```

Comment:
`allSettled` instead of `all`

Reply:
Added a helper to keep types nicely and match the error log behavior of sequential awaits, while keeping the benefits of Promise.all without the potential unhandled rejections.